File: iosApp/Views/Settings/LanguageSection.swift
import SwiftUI

/// Language picker section (spec §11). Editable, but the edit screen
/// triggers a fresh biometric prompt — spec §11 calls this out
/// explicitly to "prevent stranger-flips".
struct LanguageSection: View {

    @Environment(SettingsStore.self) private var settings

    var body: some View {
        Section {
            NavigationLink {
                EditLanguageView()
            } label: {
                HStack {
                    Text("Language")
                    Spacer()
                    Text(displayName(for: settings.language))
                        .foregroundStyle(.secondary)
                }
            }
        } header: {
            Text("Language")
        } footer: {
            Text("Changing language requires a fresh biometric prompt.")
        }
    }

    private func displayName(for raw: String) -> String {
        OnboardingLanguage(rawValue: raw)?.displayName ?? raw
    }
}

File: androidApp/src/main/kotlin/com/malaria/android/ui/settings/LanguageSection.kt
package com.malaria.android.ui.settings

import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.malaria.android.data.OnboardingLanguage
import com.malaria.android.ui.locals.LocalSettingsStore

/**
 * Language picker section (spec §11). Editable; the edit screen triggers
 * a fresh biometric prompt — spec §11 calls this out explicitly to prevent
 * stranger-flips.
 */
@Composable
fun LanguageSection(navigator: SettingsNavigator) {
    val settings = LocalSettingsStore.current
    val language by settings.language.collectAsStateWithLifecycle()

    SectionScaffold(
        header = "Language",
        footer = "Changing language requires a fresh biometric prompt.",
    ) {
        EditableRow(
            label = "Language",
            value = OnboardingLanguage.fromCanonical(language).displayName,
            enabled = true,
            onClick = { navigator.push(SettingsDestination.EditLanguage) },
        )
    }
}
